core/hash_test.go

```diff
@@ -0,0 +1,115 @@
+package core
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+func TestPedersenHash(t *testing.T) {
+	tests := []struct {
+		a, b string
+		want string
+	}{
+		{
+			"0x03d937c035c878245caf64531a5756109c53068da139362728feb561405371cb",
+			"0x0208a0a10250e382e1e4bbe2880906c2791bf6275695e02fbbc6aeff9cd8b31a",
+			"0x030e480bed5fe53fa909cc0f8c4d99b8f9f2c016be4c41e13a4848797979c662",
+		},
+		{
+			"0x58f580910a6ca59b28927c08fe6c43e2e303ca384badc365795fc645d479d45",
+			"0x78734f65a067be9bdb39de18434d71e79f7b6466a4b66bbd979ab9e7515fe0b",
+			"0x68cc0b76cddd1dd4ed2301ada9b7c872b23875d5ff837b3a87993e0d9996b87",
+		},
+	}
+	for _, tt := range tests {
+		t.Run("TestHash", func(t *testing.T) {
+			a, _ := new(felt.Felt).SetString(tt.a)
+			b, _ := new(felt.Felt).SetString(tt.b)
+			ans := new(felt.Felt)
+			want, _ := new(felt.Felt).SetString(tt.want)
+			err := PedersenHash(a, b, ans)
+			if err != nil || !ans.Equal(want) {
+				t.Errorf("TestHash got %s, want %s", ans.Text(16), want.Text(16))
+			}
+		})
+	}
+}
+
+func TestArrayDigest(t *testing.T) {
+	tests := [...]struct {
+		input []string
+		want  string
+	}{
+		// Contract address calculation. See the following links for how the
+		// it is carried out and the result referenced.
+		//
+		//	- https://docs.starknet.io/docs/Contracts/contract-address/
```

Comment:
This is a dead link. Returns `Page not found`

```suggestion
		//	- https://docs.starknet.io/documentation/develop/Contracts/contract-address/
```